test/storage_wipe_test.go

```diff
@@ -0,0 +1,272 @@
+package test
+
+import (
+	goctx "context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	asdbv1beta1 "github.com/aerospike/aerospike-kubernetes-operator/api/v1beta1"
+	"github.com/aerospike/aerospike-kubernetes-operator/pkg/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type UnwipedVolumesError struct {
+	err string
+}
+
+func (e UnwipedVolumesError) Error() string {
+	return fmt.Sprintf("unwiped volumes: %s\n", e.err)
+}
+
+var _ = Describe(
+	"StorageWipe", func() {
+		ctx := goctx.TODO()
+		Context(
+			"When doing valid operations", func() {
+
+				trueVar := true
+
+				containerName := "tomcat"
+				podSpec := asdbv1beta1.AerospikePodSpec{
+					Sidecars: []corev1.Container{
+						{
+							Name:  containerName,
+							Image: "tomcat:8.0",
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 7500,
+								},
+							},
+						},
+					},
+				}
+
+				clusterName := "storage-wipe"
+				clusterNamespacedName := getClusterNamespacedName(
+					clusterName, namespace,
+				)
+
+				It(
+
+					"Should validate all storage-wipe policies", func() {
+
+						var rackStorageConfig asdbv1beta1.AerospikeStorageSpec
+						rackStorageConfig.BlockVolumePolicy.WipeMethod = asdbv1beta1.AerospikeVolumeWipeMethodBlkdiscard
+						rackStorageConfig.BlockVolumePolicy.InitMethod = asdbv1beta1.AerospikeVolumeInitMethodDD
+						if cloudProvider == CloudProviderAWS {
+							rackStorageConfig.BlockVolumePolicy.InitMethod = asdbv1beta1.AerospikeVolumeInitMethodDD
+							rackStorageConfig.BlockVolumePolicy.WipeMethod = asdbv1beta1.AerospikeVolumeWipeMethodBlkdiscard
+						}
+						racks := []asdbv1beta1.Rack{
+							{
+								ID:      1,
+								Storage: rackStorageConfig,
```

Comment:
This storage will replace the primary storage for the rack. This doesn't have any volume so setting a rack-level storage policy will not make any difference.

Reply:
rcak ID: 2 has its own storage configuration so it can be easily configured to use other storage policy configurations if needed - the following test just focuses on testing the wiping process.